Core/include/Acts/Propagator/Propagator.ipp

```diff
@@ -44,6 +44,10 @@ auto Acts::Propagator<S, N>::propagate(propagator_state_t& state) const
       // Pre-Stepping: target setting
       state.stage = PropagatorStage::preStep;
       m_navigator.preStep(state, m_stepper);
+      if (state.options.abortList(state, m_stepper, m_navigator, logger())) {
+        terminatedNormally = true;
+        break;
+      }
```

Comment:
Thinking about this again I believe this is necessary because we switch portal in the `preStep` call instead of the `postStep` one. Should we rather make this sound between `Gen1` and `Gen2` navigation to keep the same logic in the propagation loop?

Reply:
I think we can do this, let me check ...

Reply:
Worked. I switched the code block for the portal update to the `postStep` & could remove the additional lines in the `Propagator` again.